File: packages/server/src/schemas/completion_check.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class CompletionCheckCreate(BaseModel):
    workorder_id: int
    result: str = Field(..., pattern="^(PASS|FAIL|REWORK)$")
    check_items: list[dict[str, Any]] = []
    completion_photo_url: str = ""
    reference_photo_url: str = ""
    similarity_score: float = Field(default=0.0, ge=0, le=1)
    defects: str = ""


class CompletionCheckResponse(BaseModel):
    id: int
    workorder_id: int
    result: str
    check_items: list[dict[str, Any]] | None = None
    completion_photo_url: str
    reference_photo_url: str
    similarity_score: float
    defects: str
    checked_at: datetime

    model_config = {"from_attributes": True}


class CompletionCheckStats(BaseModel):
    total: int = 0
    pass_count: int = 0
    fail_count: int = 0
    rework_count: int = 0
    pass_rate: float = 0.0


class CompletionCheckListResponse(BaseModel):
    items: list[CompletionCheckResponse]
    total: int

File: packages/web/src/api/completionCheck.ts
import api from "./index";

export interface CompletionCheckItem {
  id: number;
  workorder_id: number;
  result: string;
  check_items: Record<string, unknown>[] | null;
  completion_photo_url: string;
  reference_photo_url: string;
  similarity_score: number;
  defects: string;
  checked_at: string;
}

export interface CompletionCheckStats {
  total: number;
  pass_count: number;
  fail_count: number;
  rework_count: number;
  pass_rate: number;
}

export interface CompletionCheckListResult {
  items: CompletionCheckItem[];
  total: number;
}

export const completionCheckApi = {
  list: (params?: { workorder_id?: number; result?: string; skip?: number; limit?: number }) =>
    api.get<CompletionCheckListResult>("/completion-check/", { params }),
  create: (data: Omit<CompletionCheckItem, "id" | "checked_at">) =>
    api.post<CompletionCheckItem>("/completion-check/", data),
  get: (id: number) => api.get<CompletionCheckItem>(`/completion-check/${id}`),
  byWorkorder: (workorderId: number) =>
    api.get<CompletionCheckItem[]>(`/completion-check/workorder/${workorderId}`),
  stats: (params?: { workorder_id?: number }) =>
    api.get<CompletionCheckStats>("/completion-check/stats", { params }),
};
